libcontainer/cgroups/fs2/fs2.go

```diff
@@ -269,3 +269,32 @@ func (m *manager) OOMKillCount() (uint64, error) {
 
 	return c, err
 }
+
+func CheckMemoryUsage(dirPath string, r *configs.Resources) error {
+	if !r.MemoryCheckBeforeUpdate {
+		return nil
+	}
+
+	if r.Memory <= 0 && r.MemorySwap <= 0 {
+		return nil
+	}
+
+	usage, err := fscommon.GetCgroupParamUint(dirPath, "memory.current")
+	if err != nil {
```

Comment:
This check is on best-effort basis, so if we can't read the current usage (cgroup not yet created, or any other error), we should not fail.

Reply:
Worth adding this as a comment?